skfda/misc/scoring.py

```diff
@@ -412,6 +446,20 @@ def _mean_absolute_error_fdatagrid(
     return _multioutput_score_grid(error, multioutput)
 
 
+@mean_absolute_error.register  # type: ignore[attr-defined, misc]
+def _mean_absolute_error_fdatairregular(
+    y_true: FDataIrregular,
+    y_pred: FDataIrregular,
+    *,
+    sample_weight: Optional[NDArrayFloat] = None,
```

Comment:
I like `Optional`. Is `| None` preferred?

Reply:
The pipe operator is preferred instead of `Union` and `Optional` in modern code, as the annotations are then less verbose and have less nested objects.